src/main/java/org/folio/rest/impl/AccountsAPI.java

```diff
@@ -103,24 +103,21 @@ private CQLWrapper getCQL(String query, int limit, int offset) throws CQL2PgJSON
         .collect(Collectors.toList()))
         .map(ctx::withHoldings))
       .compose(ctx -> {
-        accounts.forEach(account -> {
-          Optional<Item> item = ctx.items.getItems().stream()
-            .filter(i -> account.getItemId().equals(i.getId()))
-            .findAny();
-
-          Optional<HoldingsRecord> holding = Optional.empty();
-          if (item.isPresent() && item.get().getHoldingsRecordId() != null) {
-            holding = ctx.holdings.getHoldingsRecords().stream()
-              .filter(h -> item.get().getHoldingsRecordId().equals(h.getId()))
-              .findAny();
-          }
-
-          String holdingRecordsId = item.map(Item::getHoldingsRecordId).orElse("");
-          String instanceId = holding.map(HoldingsRecord::getInstanceId).orElse("");
-
-          account.setHoldingsRecordId(holdingRecordsId);
-          account.setInstanceId(instanceId);
-        });
+        accounts.stream()
+          .filter(a -> a.getItemId() != null)
+          .forEach(account ->
+            ctx.items.getItems().stream()
+              .filter(item -> item.getHoldingsRecordId() != null)
+              .filter(item -> item.getId().equals(account.getItemId()))
+              .findAny()
+              .flatMap(item -> ctx.holdings.getHoldingsRecords().stream()
```

Comment:
Is that possible to extract it to separate method? From my perspective the stream inside the other stream confuses a bit